src/main/java/net/backupcup/mcde/util/EnchantmentSlot.java

```diff
@@ -59,35 +59,48 @@ public List<Choice> choices() {
     }
 
     public class Choice {
-        private Slots slot;
+        private Slots choiceSlot;
         private Identifier enchantment;
 
-        public Choice(Slots slot, Identifier enchantment) {
-            this.slot = slot;
+        public Choice(Slots choiceSlot, Identifier enchantment) {
+            this.choiceSlot = choiceSlot;
             this.enchantment = enchantment;
         }
-        public Slots getSlot() {
-            return slot;
+        public EnchantmentSlot getSlot() {
+            return EnchantmentSlot.this;
+        }
+        public Slots getChoiceSlot() {
+            return choiceSlot;
         }
         public int ordinal() {
-            return slot.ordinal();
+            return choiceSlot.ordinal();
         }
         public Identifier getEnchantmentId() {
             return enchantment;
         }
         public Enchantment getEnchantment() {
             return Registries.ENCHANTMENT.get(enchantment);
         }
+        public int getLevel() {
+            return 1;
+        }
+        public boolean isMaxedOut() {
+            return false;
+        }
+        public boolean isChosen() {
+            return false;
+        }
     }
 
-    public class ChoiceWithLevel extends Choice {
+    public class Chosen extends Choice {
```

Comment:
Consider moving `Choice` to an `interface` rather than a nested class and renaming to `Chosen` whilst renaming the `Chosen` class to `Choice`

Reply:
Okay. I didn't like this part anyway. But what I would do is to make all of it just a one class, named `Choice`, and it would be used everywhere.